src/main/java/org/icgc/argo/program_service/converter/ProgramConverter.java

```diff
@@ -176,6 +176,10 @@ default InviteUserResponse inviteIdToInviteUserResponse(@NonNull UUID inviteId)
       .build();
   }
 
+  default JoinProgramResponse toJoinProgramResponse(EgoUser egoUser){
+    return JoinProgramResponse.newBuilder().setUser(egoUserToUser(egoUser)).build();
+  }
+
```

Comment:
this can be replaced with this:
```java
  //NOTE: this is a hack for a bug in mapstruct when mapping an iterable to a wrapper (non-iterable) 
  // https://github.com/mapstruct/mapstruct/issues/607#issuecomment-309547739

  @Mapping(target = "mergeFrom", ignore = true)
  @Mapping(target = "clearField", ignore = true)
  @Mapping(target = "clearOneof", ignore = true)
  @Mapping(target = "mergeUser", ignore = true)
  @Mapping(target = "unknownFields", ignore = true)
  @Mapping(target = "mergeUnknownFields", ignore = true)
  @Mapping(target = "allFields", ignore = true)
  @Mapping(target = "user", source = "egoUser")
  JoinProgramResponse egoUserToJoinProgramResponse(Integer dummy, EgoUser egoUser);

  // Use this in your service layer
  default JoinProgramResponse egoUserToJoinProgramResponse(EgoUser egoUser){
    return egoUserToJoinProgramResponse(0, egoUser);
  }
```

The benefit of doing it this way, is that if JoinProgramResponse.proto is updated with a new field, this mapping will fail at compile time, which is much better than a silent bug or failing at runtime